project_forecast_line/models/forecast_line.py

```diff
@@ -159,15 +183,24 @@ def _convert_forecast(self, data):
 
     @api.depends("employee_resource_consumption_ids.forecast_hours", "forecast_hours")
     def _compute_consolidated_forecast(self):
-        data = {}
-        for d in self.env["forecast.line"].read_group(
-            [("employee_resource_forecast_line_id", "in", self.ids)],
-            fields=["forecast_hours"],
-            groupby=["employee_resource_forecast_line_id"],
-        ):
-            data[d["employee_resource_forecast_line_id"][0]] = d["forecast_hours"]
+        """
+        The idea is to use same computed method for consolidated forecast fields,
+        for possible forecast lines type extending.
+        Here we compute consolidated forecast only for confirmed forecast lines
+        and both forecast and confirmed forecast line types
+        """
+
+        confirmed_lines_values = self._get_resource_forecast_line_values()
+        self._compute_employee_forecast_line_id(
+            consumption_states=lambda self: self._get_consumption_states()
+            + ("forecast",)
+        )
+        consumed_lines_values = self._get_resource_forecast_line_values()
         for rec in self:
-            rec.consolidated_forecast = rec._convert_forecast(data)
+            rec.confirmed_consolidated_forecast = rec._convert_forecast(
+                confirmed_lines_values
+            )
+            rec.consolidated_forecast = rec._convert_forecast(consumed_lines_values)
```

Comment:
This needs rewriting:

1. change the _convert_forecast method to a method converting hours to days, taking a single argument and returning a single value
2. move back the logic that has been moved in that method to get the positive value for consumption lines and the sum for employee.forecast.role related lines
3. you have in grouped_line_values a dictionary which maps an employee.forecast.role line to a dictionary which has {'confirmed': total_hours, 'forecast': total_hours} and possibly one of the key is missing -> use this to compute both fields for the forecast lines. For the other lines the value is still the opposite of the consumption.